flecs/cell_population.py

```diff
@@ -16,41 +16,53 @@
 
 class CellPopulation(ABC, torch.nn.Module):
     def __init__(self, interaction_graph, n_cells=1, per_node_state_dim=1):
-        """A population of independnet cells (no cell-cell interactions).
+        """A population of independent cells (no cell-cell interactions).
 
         Args:
             interaction_graph ():
             n_cells (int): Number of independent cells in the population.
         """
         super().__init__()
         # str type of node (e.g., gene, protein).
-        self._node_set_dict: Dict[str, NodeSet] = {}
+        self._node_set_dict: Dict[str, sets.NodeSet] = {}
         # str types of interactions (src, interaction_type, dest).
-        self._edge_set_dict: Dict[Tuple[str, str, str], EdgeSet] = {}
+        self._edge_set_dict: Dict[Tuple[str, str, str], sets.EdgeSet] = {}
 
         self.initialize_from_interaction_graph(interaction_graph)
 
-        self.state = 10 * torch.ones((n_cells, self.n_nodes, per_node_state_dim))
+        # Create state and production/decay rates as empty tensors
+        self.state = torch.empty((n_cells, self.n_nodes, per_node_state_dim))
         self.decay_rates = torch.empty((n_cells, self.n_nodes, per_node_state_dim))
         self.production_rates = torch.empty((n_cells, self.n_nodes, per_node_state_dim))
 
+        # Initialize
+        self.reset_state()
+
+    def sample_from_state_prior_dist(self):
+        return 10 * torch.ones(self.state.shape)
```

Comment:
```
SCALE_FACTOR = 10  # Arbitrary to prevent prior being all 1s (which would have no effect).
return SCALE_FACTOR * torch.ones(self.state.shape)
```